client/verify_test.go

```diff
@@ -5,50 +5,59 @@ import (
 	"fmt"
 	"testing"
 
+	"github.com/drand/drand/utils"
+
 	"github.com/drand/drand/client"
 	"github.com/drand/drand/client/test/result/mock"
 )
 
-func mockClientWithVerifiableResults(n int) (client.Client, []mock.Result, error) {
-	info, results := mock.VerifiableResults(n)
+func mockClientWithVerifiableResults(n int, decouplePrevSig bool) (client.Client, []mock.Result, error) {
+	info, results := mock.VerifiableResults(n, decouplePrevSig)
 	mc := client.MockClient{Results: results, StrictRounds: true}
-	c, err := client.Wrap(
-		[]client.Client{client.MockClientWithInfo(info), &mc},
-		client.WithChainInfo(info),
-		client.WithVerifiedResult(&results[0]),
-		client.WithFullChainVerification(),
-	)
+
+	var c client.Client
+	var err error
+	if decouplePrevSig {
+		c, err = client.Wrap(
+			[]client.Client{client.MockClientWithInfo(info), &mc},
+			client.WithChainInfo(info),
```

Comment:
This line should be enough by itself, because in the chain info there should be the version tag that indicates how to verify beacon, we shouldn't need to separate the two clients  via this bool parameters.